3684/1/options_parser.rb

```diff
@@ -0,0 +1,40 @@
+require 'optparse'
+
+class OptionsParser
+  attr_reader :flags
+
+  def initialize(file_string)
+    @flags = {}
+    @flags[:file_name] = file_string.to_s
```

Comment:
from this perspective you can do:
`@flags = { file_name: file_string.to_s }`

Reply:
but since you will update this hash you can move this to the method `parse_options` which gonna be called in the beginning of method `parse`